File: contains_dup/contains_dup.js
const containsDups = (nums) => {
  let counted = {}

  for(i = 0; i < nums.length; i++){
    if(counted[nums[i]]){
      return true
    } else {
      counted[nums[i]] = true
    }
  }
  return false
}


console.log(containsDups([1,2,3,1])) // true
console.log(containsDups([1,2,3,4])) // false 

File: contains_dup/contains_dup.rb
def contains_dup(nums)
  nums_counted = {}

  nums.each do |num|
    if nums_counted[num] 
      return true
    else
      nums_counted[num] = true
    end
  end
  false
end


p contains_dup([1,2,3,1]) #true
p contains_dup([1,2,3,4]) #false
